tests/functional/testplan/runnable/interactive/test_api.py

```diff
@@ -470,6 +479,7 @@ def test_run_testcase(plan):
             raise_on_timeout=True,
         )
 
+# TODO: needs a bit more re-org around checking state transition
```

Comment:
reorg how?

Reply:
nothing, just a note to myself

Reply:
if you are planning to reorg this test module can we discuss it first please

Reply:
not in this PR